learn/framework/xcm.md

```diff
@@ -0,0 +1,73 @@
+
+## Design Principles {: #design-principles }
+
+Conceived with an abstract mindset, XCM is not coupled with either a specific use case or a specific destination appchain setup, thus minimizing the coupling effect. Its core design principles are:
+
+- **Asynchronous** - similar to sending a postcard -but way faster- the sender will keep performing its duties as usual, without blocking itself or awaiting a response from the destination
+- **Absolute** -  messages are guaranteed to be delivered to the intended destination, in order and in a timely fashion
+- **Asymmetric** -  messages sent have no response counterpart. Any return values, if required, must be sent back from the destination to the sender with another message
+- **Agnostic** -  there are no assumptions whatsoever about the configuration or properties of the appchains that are communicating. Two communicating appchains might differ in every aspect, except the ability to understand XCM. E.g., one chain could be EVM-compatible and the other not, one chain could be a DeFi appchain and the other a gaming appchain, etc
+
+## Common Use Cases {: #common-use-cases }
+
+Benefiting from the common ground and versatility XCM provides, many use cases can be addressed. Two of the most recurrent ones are asset transfers and remote execution.
+
+### Asset Transfers {: #asset-transfer }
+
+Moving digital assets from one appchain to another is essential for creating a more dynamic, efficient, and interconnected blockchain ecosystem. The native cross-chain capability allows two main strategies to transfer assets from one chain to another:
+
+- **Teleport** - teleporting an asset is a simple and efficient mechanism, but it has a major caveat: it requires trust between the parties. In essence, when appchain A wants to send X amount of the assets to appchain B, it sends a message communicating appchain B that they must mint exactly X amount of the assets. After relaying the message, appchain A burns X amount of the assets, keeping the overall asset balance and concluding the teleport action
+- **Reserve transfer** - a reserve transfer involves a trusted third party acting as a reserve of value. When appchain A wants to send X amount of an asset to appchain B, it sends the instruction to the reserve, who transfers the assets from appchain A to appchain B reserve accounts. The process is completed when appchain A burns X amount of the assets and appchain B mints them. Note that regardless of appchain A not burning the assets or appchain B minting tokens in excess, since the wrongful tokens are not backed one-to-one by the reserve, they have no real value
+
+### Remote Execution {: #remote-execution }
+
+Another common use is remote execution, which is sending a message to an appchain and triggering some action. If the destination chain is an EVM-compatible one, this allows appchain A to call a smart contract deployed on appchain B, for example.
```

Comment:
I would expand this a bit. Expand on what is the general flow.

Some thing are similar between asset transfer and remote execution that you never touch upon. For example, the need of XCM delivery fees (which should be cover here) and also the need of buying execution in the destination chain